backend/infrahub/core/query/relationship.py

```diff
@@ -340,6 +340,11 @@ async def query_init(self, db: InfrahubDatabase, **kwargs) -> None:  # noqa: ARG
             r1,
             r2,
         )
+        if self.branch.is_default or self.branch.is_global:
```

Comment:
Just noting that we seem to use `if self.branch.is_default or self.branch.is_global:` in a number of places. Might be worth considering having a property for the same thing on the branch objects.